packages/headless/src/controllers/product-listing/headless-product-listing.ts

```diff
@@ -0,0 +1,61 @@
+import {executeProductListingSearch} from '../../features/product-listing/product-listing-actions';
+import {buildController} from '../controller/headless-controller';
+import {Schema, SchemaValues} from '@coveo/bueno';
+import {configuration, productListing} from '../../app/reducers';
+import {loadReducerError} from '../../utils/errors';
+import {ProductListingEngine} from '../../app/product-listing-engine/product-listing-engine';
+
+// TODO COM-1185 - Actually add options and dispatch the updates
+const optionsSchema = new Schema({});
+
+export type ProductListingListOptions = SchemaValues<typeof optionsSchema>;
+
+export interface ProductListingListProps {
+  options?: ProductListingListOptions;
+}
+
+export type ProductListingList = ReturnType<typeof buildProductListing>;
+export type ProductListingListState = ProductListingList['state'];
+
+export const buildProductListing = (
+  engine: ProductListingEngine,
+  _props: ProductListingListProps = {}
+) => {
+  if (!loadBaseProductListingReducers(engine)) {
+    throw loadReducerError;
+  }
+
+  const controller = buildController(engine);
+  const {dispatch} = engine;
+  const getState = () => engine.state;
+
+  // TODO COM-1185 - Actually add options and dispatch the updates
+  /*const options = validateOptions(
```

Comment:
A pattern I found helpful was to separate the building of the values from the validation. It should make this TODO easier to conquer :)

```
const options = {
    ...defaultOptions,
    ...props.options,
}

validate(options) // only using validation, returned value and type are ignored
```